modules/peridynamics/src/meshgenerators/MeshGeneratorPD.C

```diff
@@ -21,97 +21,111 @@ InputParameters
-    mooseError("Please specifiy either 'convert_block_ids' or 'non_convert_block_ids'!");
+  if (_has_blks_to_pd && _has_blks_as_fe)
+    mooseError("Please specifiy either 'blocks_to_pd' or 'blocks_as_fe'!");
 
-  if (_has_non_conv_blk_ids)
+  if (_has_blks_as_fe)
   {
-    std::vector<SubdomainID> ids = getParam<std::vector<SubdomainID>>("non_convert_block_ids");
+    std::vector<SubdomainID> ids = getParam<std::vector<SubdomainID>>("blocks_as_fe");
     for (unsigned int i = 0; i < ids.size(); ++i)
-      _non_conv_blk_ids.insert(ids[i]);
+      _blks_as_fe.insert(ids[i]);
   }
 
-  if (_has_conv_blk_ids)
+  if (_has_blks_to_pd)
   {
-    std::vector<SubdomainID> ids = getParam<std::vector<SubdomainID>>("convert_block_ids");
+    std::vector<SubdomainID> ids = getParam<std::vector<SubdomainID>>("blocks_to_pd");
     for (unsigned int i = 0; i < ids.size(); ++i)
-      _conv_blk_ids.insert(ids[i]);
+      _blks_to_pd.insert(ids[i]);
   }
 
-  if (_has_connect_blk_id_pairs && _has_non_connect_blk_id_pairs)
-    mooseError("Please specifiy either 'connect_block_id_pairs' or 'non_connect_block_id_pairs'!");
+  if (!_construct_pd_sideset && _has_sidesets_to_pd)
+    mooseError("'sidesets_to_pd' is provided without setting "
+               "'construct_pd_sidesets' to 'true'!");
 
-  _connect_blk_id_pairs.clear();
-  if (_has_connect_blk_id_pairs)
+  if (_has_sidesets_to_pd)
   {
-    std::vector<SubdomainID> ids = getParam<std::vector<SubdomainID>>("connect_block_id_pairs");
-
-    if (ids.size() % 2 != 0)
-      mooseError("Input parameter 'connect_block_id_pairs' must contain even number of entries!");
-
-    const unsigned int pairs = ids.size() / 2;
-    for (unsigned int i = 0; i < pairs; ++i) // consider the renumbering of IDs of converted blocks
-      _connect_blk_id_pairs.insert(std::make_pair(ids[2 * i] + 1000, ids[2 * i + 1] + 1000));
+    std::vector<boundary_id_type> ids = getParam<std::vector<boundary_id_type>>("sidesets_to_pd");
+    for (unsigned int i = 0; i < ids.size(); ++i)
+      _fe_sidesets_for_pd_construction.insert(ids[i]);
   }
 
-  _non_connect_blk_id_pairs.clear();
-  if (_has_non_connect_blk_id_pairs)
+  if (_has_bonding_blk_pairs && _has_non_bonding_blk_pairs)
+    mooseError("Please specifiy either 'bonding_block_pairs' or "
+               "'non_bonding_block_pairs'!");
+
+  _pd_bonding_blk_pairs.clear();
+  if (_has_bonding_blk_pairs)
   {
-    std::vector<SubdomainID> ids = getParam<std::vector<SubdomainID>>("non_connect_block_id_pairs");
+    std::vector<std::vector<SubdomainID>> id_pairs =
+        getParam<std::vector<std::vector<SubdomainID>>>("bonding_block_pairs");
 
-    if (ids.size() % 2 != 0)
-      mooseError(
-          "Input parameter 'non_connect_block_id_pairs' must contain even number of entries!");
+    for (unsigned int i = 0; i < id_pairs.size(); ++i)
+      _pd_bonding_blk_pairs.insert(std::make_pair(id_pairs[i][0] + 1000, id_pairs[i][1] + 1000));
```

Comment:
I'd suggest creating a variable called `block_offset` or something like that and setting that to 1000, and then adding `block_offset` here. That would make it more clear what that 1000 means.